crates/sui/tests/onsite_reconfig_observer_tests.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use prometheus::Registry;
+use sui_core::authority_aggregator::AuthAggMetrics;
+use sui_core::quorum_driver::reconfig_observer::OnsiteReconfigObserver;
+use sui_core::quorum_driver::reconfig_observer::ReconfigObserver;
+use sui_core::safe_client::SafeClientMetricsBase;
+use test_utils::authority::{spawn_test_authorities_with_fullnodes, test_authority_configs};
+use test_utils::network::wait_for_nodes_transition_to_epoch;
+
+#[tokio::test]
```

Comment:
i would suggest using sim_test for determinism

Reply:
see https://github.com/MystenLabs/sui/pull/7024#discussion_r1061972678